CHANGELOG.md

```diff
@@ -12,6 +12,7 @@
 * ECS - Support affiliations select for the inventory field (`<ConsortiumFieldInventory>`). Refs UISACQCOMP-190.
 * Add Print routing list functionality. Refs UISACQCOMP-191.
 * ECS - Support central ordering in the acq modules. Refs UISACQCOMP-194.
+* *BREAKING* `useFilters` - remove search term trimming. Refs UISACQCOMP-198.
```

Comment:
Hey @zburke can we call this a breaking change? If I understand correctly technically no dependency upgrades are needed and public API of the module hasn't changed, so no need to bump up the major version, right?

Reply:
Two things are happening here: 
1. fixing Firefox's cursor shift feels like a bug fix; this change removes unexpected behavior that nobody should be relying on
2. removing `trim()` feels like a breaking change**; the change removes expected behavior that we know some folks (Orders) rely on

Can you fix 1 without removing 2? The bug is easy to reproduce, even in Chrome, but also seems easy to fix by calling `setSelectionRange()` to restore the previous cursor position. Run this as-is to see the bug, and uncomment the last line to fix it. 
```
<html>
<head>
</head>
<body>
<div><input id="foo" /></div>

<script>
document.onkeyup = (_e) => {
  const foo = document.getElementById('foo');
  const pos = foo.selectionDirection == 'backward' ? foo.selectionStart : foo.selectionEnd;

  foo.value = foo.value.trim();
//  foo.setSelectionRange(pos, pos);
}
</script>
</body>
</html>
```

** Breaking changes can be due to many reasons, e.g. requiring a new API, removing a public export, adding a new required argument, changing the behavior of an existing function, etc. True, this PR has no dependency changes or public API changes, but it is clear that the behavior of an existing function is changing.

Reply:
Hi @zburke, thank you for providing this solution. However, I've noticed that the user expects spaces, but they are being removed.

@zburke, @BogdanDenis, @NikitaSedyx, @usavkov-epam - I believe addressing this bug in the components module will benefit all modules that rely on this functionality, not just `ui-inventory`. By resolving this issue globally, we can prevent similar problems from occurring in the future. Although I could fix this only in the `ui-inventory` module, I think it would be more effective and maintainable to resolve it at the source.

I'm also willing to fix the search in the `ui-orders` module. I'm open to suggestions and finding the best solution. What do you think would be the best way to proceed?

Reply:
@Dmytro-Melnyshyn , I have no skin in the game here. If you think most consumers of this API expect the leading/trailing spaces to remain as-is (and you can handle ui-orders separately), then removing `.trim()` and publishing this as a breaking changes is fine.